### PR TITLE
refactor(sdf): separate out `resources.py`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Basically the same as https://github.com/dagster-io/dagster/pull/23154, https://github.com/dagster-io/dagster/pull/23161, and https://github.com/dagster-io/dagster/pull/23209, but for `dagster-sdf`.

## How I Tested These Changes
bk

### CODE REVIEW

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23239?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23239** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23239?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23238** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23238?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23239?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23239** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23239?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23238** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23238?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-sdf/dagster_sdf/__init__.py [hunk 1]
```diff
@@ -6,9 +6,9 @@
     dagster_name_fn as dagster_name_fn,
     default_asset_key_fn as default_asset_key_fn,
 )
-from .information_schema import SdfInformationSchema as SdfInformationSchema
 from .resource import SdfCliResource as SdfCliResource
+from .sdf_information_schema import SdfInformationSchema as SdfInformationSchema
+from .sdf_workspace import SdfWorkspace as SdfWorkspace
```
Comment (cmpadden) [inline python_modules/libraries/dagster-sdf/dagster_sdf/__init__.py:11]: Appears to have changes included in https://github.com/dagster-io/dagster/pull/23238/files

**Review by cmpadden — APPROVED** [review]
fabuloso - pending bk

**Comment by rexledesma** [comment]
### Merge activity

* **Jul 25, 7:26 PM PDT**: @rexledesma merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/23239).

**Pull request merged**